src/app/modules/shared/components/details-fields/details-fields.component.ts

```diff
@@ -264,12 +264,37 @@ export class DetailsFieldsComponent implements OnChanges, OnInit {
     return startDate >= endDate;
   }
 
-  dateToSubmit(date, hour) {
-    const entryFormDate = this.entryForm.value[date];
-    const updatedHour = this.entryForm.value[hour];
-    const updatedDate = new Date(`${entryFormDate}T${updatedHour.trim()}`).toISOString();
+  getDateISOFormat(date: string, hourAndMinutes: string, seconds, miliseconds: number): string {
+    hourAndMinutes = hourAndMinutes.trim();
+    const secondsInISOFormat: string = this.getNumberInISOFornat(seconds);
+    const milisecondsInISOFormat: string = this.getNumberInISOFornat(miliseconds);
+    const ISOFormat = `${date}T${hourAndMinutes}:${secondsInISOFormat}.${milisecondsInISOFormat}`;
+    return new Date(ISOFormat).toISOString();
+  }
+
+  getNumberInISOFornat(numberToFormat: number): string {
+    const limitSingleNumber = 9;
+    const isNumberGreaterThanLimitNumber = numberToFormat > limitSingleNumber;
+    return isNumberGreaterThanLimitNumber ? numberToFormat.toString() : `0${numberToFormat}`;
+  }
+
+  dateToSubmit(date: string, hour: string) {
+    const timeEntryISODate: string = get(this.entryToEdit, date);
+    let seconds = 0;
+    let miliseconds = 0;
```

Comment:
Set the types in this constants `seconds, miliseconds`

Reply:
Same as above